pkg/authentication/auth0.go

```diff
@@ -0,0 +1,73 @@
+package authentication
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/golang-jwt/jwt/v5"
+	"strings"
+)
+
+var (
+	// ErrTokenNoProvided is returned when no token is provided.
+	ErrTokenNoProvided = errors.New("no token provided")
```

Comment:
nit: `TokenNotProvided` to make it sound a little more natural?

```suggestion
	// ErrTokenNotProvided is returned when no token is provided.
	ErrTokenNotProvided = errors.New("no token provided")
```